src/builtins/prelude.js

```diff
@@ -34,12 +37,18 @@ require = (function (modules, cache, entry) {
         err.code = 'MODULE_NOT_FOUND';
         throw err;
       }
-      
+
       localRequire.resolve = resolve;
 
       var module = cache[name] = new newRequire.Module;
 
       modules[name][0].call(module.exports, localRequire, module, module.exports);
+
+      // Expose the exports if we are on the entry point by using the
+      // previous defined module.
+      if (previousModule && name === 1) {
```

Comment:
Checking `name === 1` would break if we ever change the way IDs are generated (such as #780). Not sure how else this can be checked, though. @devongovett  ?

Reply:
Yeah. Also in split bundles, the entry module does not have `id === 1`. You can look in the `entry` argument passed to this function above to get the entry module for the bundle. It is an array, but the last element will be the id of entry module. See here: https://github.com/parcel-bundler/parcel/blob/595252c4135c9f9d6cbb8aa5f3f2e94771cc9712/src/packagers/JSPackager.js#L197-L199.